lib/models/device.coffee

```diff
@@ -275,23 +275,20 @@ exports.identify = (uuid, callback) ->
 # @param {String} newName - the device new name
 # @param {module:resin.models.device~renameCallback} callback - callback
 #
-# @todo This action doesn't return any error
-# if trying to rename a device that does not
-# exists. This should be fixed server side.
-#
 # @example
 #	resin.models.device.rename 317, 'NewName', (error) ->
 #		throw error if error?
 #		console.log("Device has been renamed!")
 ###
 exports.rename = (name, newName, callback) ->
-	return pine.patch
-		resource: 'device'
-		body:
-			name: newName
-		options:
-			filter:
-				name: name
+	exports.get(name).then ->
```

Comment:
Right. Well, that's a major issue. I knew device names were not unique, but I thought there couldn't be duplicated devices per user.

In that case, the only reliable way to refer to a device would be with an id, which was how the SDK worked initially, and everybody complained.

The other solution I can think of is disallowing duplicated device names from within an application. That way we can refer to devices as `<application>/<device>`, in a Github manner, for example: `resin device MyApp/MyDevice`.

Thoughts?